src/build-artifact/run.ts

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs';
+import path from 'path';
+import readline from 'readline';
+import { Readable } from 'stream';
+import { finished } from 'stream/promises';
+import type * as streamWeb from 'node:stream/web';
+
+/**
+  Typescript necessary declaration:
+  https://stackoverflow.com/questions/73308289/typescript-error-converting-a-native-fetch-body-webstream-to-a-node-stream
+*/
+declare global {
+  interface Response {
+    readonly body: streamWeb.ReadableStream<Uint8Array> | null;
+  }
+}
+
+// These types are what's in the snooty manifest json file.
+export type SnootyManifestEntry = {
+  type: "page" | "timestamp" | "metadata" | "asset";
+  data: unknown;
+};
+
+/**
+  A node in the Snooty AST.
+ */
+type SnootyNode = {
+  type: string;
+  children?: (SnootyNode | SnootyTextNode)[];
+  options?: Record<string, unknown>;
+  [key: string]: unknown;
+};
+
+/**
+  A Snooty AST node with a text value.
+ */
+type SnootyTextNode = SnootyNode & {
+  type: "text";
+  children: never;
+  value: string;
+};
+
+/**
+  A page in the Snooty manifest.
+ */
+type SnootyPageData = {
+  page_id: string;
+  ast: SnootyNode;
+  tags?: string[];
+  deleted: boolean;
+};
+
+/**
+ An Asset map by checksum 
+ */
+type Assets = {
+  [checksum: string]: {
+    checksum: string;
+    assetData: string;
+    filenames: string[];
+  };
+}
+
+export async function run(): Promise<void> {
+  try {
+    const file = 'output.txt'
+    /* Fetch Snooty project build data */
+    await downloadSnootyProjectBuildData(`https://snooty-data-api.mongodb.com/projects/${process.env.PROJECT_TO_BUILD}/master/documents`, file);
+
+    let metadata: SnootyManifestEntry;
+    const documents: SnootyPageData[] = [];
+    const assets: Assets = {};
+
+    /* Write each line to separate files in expected data structure for Snooty */
+    readline.createInterface({
+        input: fs.createReadStream(file),
+        terminal: false
+    }).on('line', function(lineString: string) {
```

Comment:
(Optional optimization) We can probably skip writing the API stream to a txt file by using the response body from the fetch as the input. Although I'm not sure if this is only possible with `node-fetch`, something like:

```js
import { createInterface } from "readline";
import fetch from "node-fetch";

const { body } = await fetch(apiUrl);

if (body === null) {
  throw new Error('No response body found');
}

const documents = []; // etc.
createInterface(body).on('line', () => {});
```

The frontend's preview plugin does something similar, except with `got` instead of `fetch`.